src/content/reference/device-os/firmware.md

```diff
@@ -13782,6 +14066,220 @@ bool isEmpty() const;
 See [Variant](#variant) and [Map](#map) for additional information.
 
 
+## Buffer
+
+{{api name1="Buffer"}}
+
+{{since when="6.2.0"}}
+
+The `Buffer` class is a container for data in a dynamically allocated buffer. It's essentially a wrapper
+around `std::vector<char>` with additional convenience methods for encoding the data.
+
+The `Variant` class has been extended to store arbitrary binary data in a `Buffer`. A Variant containing a `Buffer` 
+cannot be serialized to JSON or deserialized from JSON as JSON does not support binary values.
```

Comment:
Is this statement true? Won't a Variant containing a Buffer be serialized with `{ "_type": "Buffer", "_data": "11aabbee22" }`

Reply:
I thought that the Variant to JSON occurred on the cloud-side, not when using the toJSON() method of Variant, but I could have misunderstood. However, my wording doesn't make it clear that's what I meant, so I'll fix that regardless.

Reply:
> I thought that the Variant with buffer to JSON occurred on the cloud-side, not when using the toJSON() method of Variant

That's correct. `toJSON()` doesn't use the same conversion rules as the cloud does when a variant is published which may indeed cause some confusion. I'd probably just remove this paragraph about `Variant` given that the docs for `toJSON()` already mention that it can't serialize values of certain types.

I thought about using the same conversion rules both in Device OS and the cloud but wasn't keen on pulling Base64 code in Device OS.